ecdsa/src/ecdsa.rs

```diff
@@ -262,27 +283,56 @@ mod tests {
                     let offset = 0;
                     let ctx = &mut RegionCtx::new(region, offset);
 
+                    // r and s should be less than Remiander
+                    let r = if self.valid_input {
+                        let r = format!("{:?}", self.r);
+                        big_uint::from_str_radix(&r[2..], 16).unwrap()
+                    } else {
+                        scalar_chip.rns().max_remainder.clone() + big_uint::from(20u32)
+                    };
+                    let s = if self.valid_input {
+                        let s = format!("{:?}", self.s);
+                        big_uint::from_str_radix(&s[2..], 16).unwrap()
+                    } else {
+                        scalar_chip.rns().max_remainder.clone() + big_uint::from(20u32)
+                    };
+                    let is_r_s_within_ranage = scalar_chip.assign_constant(ctx, ((r <= scalar_chip.rns().max_remainder && s <= scalar_chip.rns().max_remainder) as u64).into())?;
+                    let is_valid = scalar_chip.is_not_zero_without_reduce(ctx, &is_r_s_within_ranage)?;
+
                     let r = self.signature.map(|signature| signature.0);
                     let s = self.signature.map(|signature| signature.1);
                     let integer_r = ecc_chip.new_unassigned_scalar(r);
                     let integer_s = ecc_chip.new_unassigned_scalar(s);
                     let msg_hash = ecc_chip.new_unassigned_scalar(self.msg_hash);
 
-                    let r_assigned =
-                        scalar_chip.assign_integer(ctx, integer_r, Range::Remainder)?;
-                    let s_assigned =
-                        scalar_chip.assign_integer(ctx, integer_s, Range::Remainder)?;
+                    let (r_assigned, is_assigned_integer_succeeded ) =
+                        scalar_chip.try_assign_integer(ctx, integer_r, Range::Remainder)?;
+                    let is_valid = scalar_chip.and(ctx, &is_valid, &is_assigned_integer_succeeded)?;
+                    let (s_assigned, is_assigned_integer_succeeded) =
+                        scalar_chip.try_assign_integer(ctx, integer_s, Range::Remainder)?;
+                    let is_valid = scalar_chip.and(ctx, &is_valid, &is_assigned_integer_succeeded)?;
```

Comment:
`try_assign_integer` will do the range check no? So this seems sufficient, not sure why `is_r_s_within_ranage` is used above this code?